src/Database/Auth/ApiKeyAuthenticator.ts

```diff
@@ -101,14 +102,14 @@ export class ApiKeyAuthenticator {
       let publicKey: Buffer, body: IResponseBody;
 
       if (!StatusCodes.isOk(response.statusCode)
-        && !TypeUtil.isObject(body = response.body) && !body.PublicKey
+        && !TypeUtil.isObject(body = response.body as IResponseBody) && !body.PublicKey
```

Comment:
A response from ravendb must always be an object, there shouldn't be anything that returns arrays.